src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -684,6 +714,63 @@ private static bool InitRTNoCacheSupported()
                 "NoCache");
         }
 
+        private void RTServerCertificateCallback(RTHttpBaseProtocolFilter sender, RTHttpServerCustomValidationRequestedEventArgs args)
+        {
+            // Convert WinRT certificate to .NET certificate.
+            X509Certificate2 serverCert = ConvertCertificate(args.ServerCertificate);
+
+            // Create .NET X509Chain from the WinRT information. We need to rebuild
+            // the chain since WinRT only gives us an array of intermediate certificates.
+            var serverChain = new X509Chain();
+            foreach (RTCertificate cert in args.ServerIntermediateCertificates)
+            {
+                serverChain.ChainPolicy.ExtraStore.Add(ConvertCertificate(cert));
+            }
+            serverChain.Build(serverCert);
+
+            // Determine chain and name errors from WinRT information.
+            SslPolicyErrors sslPolicyErrors = SslPolicyErrors.None;
+            if (serverChain.ChainStatus.Length > 0)
+            {
+                sslPolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+            }
+            foreach (RTChainValidationResult result in args.ServerCertificateErrors)
```

Comment:
https://github.com/dotnet/corefx/blob/81f80a8a0088d482afdb077b23f21a3e75a1c30c/src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpCertificateHelper.cs#L29-L39

You're missing the `chain.ChainPolicy.ApplicationPolicy.Add(s_serverAuthOid)` part (the OID check) as well as the RevocationMode/RevocationFlag choices (Online/ExcludeRoot are the defaults, so maybe you're getting the values you want).